mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapperTest.java

```diff
@@ -73,6 +77,23 @@ void findInMailboxLimitShouldLimitProjectionReadCassandraQueries(CassandraCluste
             .hasSize(limit);
     }
 
+    @Test
+    void updateFlagsShouldLimitModSeqAllocation(CassandraCluster cassandra) throws MailboxException {
+        saveMessages();
+
+        StatementRecorder statementRecorder = new StatementRecorder();
+        cassandra.getConf().recordStatements(statementRecorder);
+
+        messageMapper.updateFlags(benwaInboxMailbox, new FlagsUpdateCalculator(new Flags(Flags.Flag.ANSWERED), MessageManager.FlagsUpdateMode.REPLACE), MessageRange.all());
+
+        assertThat(statementRecorder.listExecutedStatements())
+            .filteredOn(statement -> statement instanceof BoundStatement)
+            .extracting(BoundStatement.class::cast)
+            .extracting(statement -> statement.preparedStatement().getQueryString())
+            .filteredOn(statementString -> statementString.equals("UPDATE modseq SET nextModseq=:nextModseq WHERE mailboxId=:mailboxId IF nextModseq=:modSeqCondition;"))
```

Comment:
a much better test would be to check the value of modseq instead.

Reply:
I disagree. I intend to test the pressure we put on underlying services, and counting queries we do exeute is a straightforward way of doing this.

Value of the modseq can have increments greater than one (in concurrent environment).

Reply:
yet this test is a transparent-box (alternative to whitebox/blackbox) one (we look at how not what), it's always more brittle and more complex to maintain than an opaque-box one.
I don't argue it has value but if I need to pick only one, I would pick the modseq one. The conclusion should be: we could have both.